cast/src/lib.rs

```diff
@@ -110,6 +110,60 @@ where
         })
     }
 
+    /// Generates an access list for the specified transaction
+    ///
+    /// ```no_run
+    /// 
+    /// use cast::Cast;
+    /// use ethers_core::types::{Address, Chain};
+    /// use ethers_providers::{Provider, Http};
+    /// use std::{str::FromStr, convert::TryFrom};
+    ///
+    /// # async fn foo() -> eyre::Result<()> {
+    /// let provider = Provider::<Http>::try_from("http://localhost:8545")?;
+    /// let cast = Cast::new(provider);
+    /// let to = Address::from_str("0xB3C95ff08316fb2F2e3E52Ee82F8e7b605Aa1304")?;
+    /// let sig = "function greeting(uint256 i) public returns (string)";
+    /// let args = vec!["5".to_owned()];
+    /// let access_list = cast.access_list(Address::zero(), to, (sig, args), Chain::Mainnet, None, false).await?;
+    /// println!("{}", access_list);
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn access_list<F: Into<NameOrAddress>, T: Into<NameOrAddress>>(
+        &self,
+        from: F,
+        to: T,
+        args: (&str, Vec<String>),
+        chain: Chain,
+        block: Option<BlockId>,
+        to_json: bool,
+    ) -> Result<String> {
+        let (tx, _) =
+            self.build_tx(from, to, Some(args), None, None, None, None, chain, None, false).await?;
```

Comment:
Think we're at the point where this should be refactored to a builder pattern struct :P Not for this PR, but generally good first issue